webapp/src/js/services/rules-engine.js

```diff
@@ -128,11 +128,10 @@ angular.module('inboxServices').factory('RulesEngine', function(
       key: 'mark-contacts-dirty',
       filter: change => !!change.doc && (ContactTypes.includes(change.doc) || isReport(change.doc)),
       callback: change => {
-        const subjectId = isReport(change.doc) ? registrationUtils.getSubjectId(change.doc) : change.id;
+        const subjectIds = isReport(change.doc) ? registrationUtils.getSubjectIds(change.doc) : [ change.id ];
         const telemetryData = telemetryEntry('rules-engine:update-emissions', true);
-
         return RulesEngineCore
-          .updateEmissionsFor(subjectId)
+          .updateEmissionsFor(subjectIds)
```

Comment:
Sorry, I was completely wrong and this isn't enough. We have to do something similar when we recalculate, after we get the reports from the db. :( Sorry